android/hello_sdl_android/src/main/java/com/sdl/hellosdlandroid/MainActivity.java

```diff
@@ -37,24 +38,53 @@ protected void onCreate(Bundle savedInstanceState) {
         }
     }
 
-    private boolean checkPermission() {
-        return PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), BLUETOOTH_CONNECT);
+    private boolean checkBTPermission() {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !checkPermission(Manifest.permission.BLUETOOTH_CONNECT);
+    }
+
+    private boolean checkPNPermission() {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && !checkPermission(Manifest.permission.POST_NOTIFICATIONS);
     }
 
-    private void requestPermission() {
-        ActivityCompat.requestPermissions(this, new String[]{BLUETOOTH_CONNECT}, REQUEST_CODE);
+    private boolean checkPermission(String permission) {
+        return PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getApplicationContext(), permission);
+    }
+
+    private void requestPermission(String[] permissions, int REQUEST_CODE) {
+        ActivityCompat.requestPermissions(this, permissions, REQUEST_CODE);
+    }
+
+    private String[] permissionsNeeded() {
```

Comment:
This isn't a big deal, but still a good idea.

```suggestion
    private @NonNull String[] permissionsNeeded() {
```